### PR TITLE
Bug 1182032 - Clean site icons cache regularly

### PR DESCRIPTION
**Pull request by gmarty**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a5d26fd44e7e4602fffe4285a7f477891ef2612a

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b01708664608835ab83451cced33231e99de572a

File: apps/system/js/browser_settings.js [hunk 1]
```diff
@@ -46,6 +46,11 @@
         request.onsuccess = function() {
           request.result.clearBrowserData();
         };
+
+        // Clear the icon dataStore.
+        LazyLoader.load('shared/js/icons_helper.js').then(() => {
+          IconsHelper.clear();
```
Comment (benfrancis) [inline apps/system/js/browser_settings.js]: We need to not clear the icons for pinned sites and pinned pages, which is going to be a bit tricky. Not sure there's anything we can do in this bug.

Comment (gmarty) [inline apps/system/js/browser_settings.js]: Let's wait until this feature is implemented.

**Pull request merged**